cdap-ui/cypress/integration/plugins.validation.spec.ts

```diff
@@ -0,0 +1,104 @@
+import {getArtifactsPoll, loginIfRequired} from '../helpers';
+import {DEFAULT_BIGQUERY_DATASET, DEFAULT_BIGQUERY_TABLE, DEFAULT_GCP_PROJECTID, DEFAULT_GCP_SERVICEACCOUNT_PATH} from '../support/constants';
+import {INodeInfo} from '../typings';
+
+const sourceNode: INodeInfo = {
+  nodeName: 'BigQueryTable',
+  nodeType: 'batchsource'
+};
+
+const sourceProperties = {
+  referenceName: 'BQ_Source',
+  project: DEFAULT_GCP_PROJECTID,
+  datasetProject: DEFAULT_GCP_PROJECTID,
+  dataset: DEFAULT_BIGQUERY_DATASET,
+  table: DEFAULT_BIGQUERY_TABLE,
+  serviceFilePath: DEFAULT_GCP_SERVICEACCOUNT_PATH,
+};
+
+let headers = {};
+
+describe('Plugin properties', () => {
+  // Uses API call to login instead of logging in manually through UI
+  before(() => {
+    loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
+        };
+      });
+    });
+    cy.visit('/pipelines/ns/default/studio');
+  });
+
+  beforeEach(() => {
+    getArtifactsPoll(headers);
+  });
+
+  it('should be validated and errors found in properties', () => {
+    cy.add_node_to_canvas(sourceNode);
+    cy.get(
+          '[data-cy="plugin-node-BigQueryTable-batchsource-0"] .node .node-configure-btn')
+        .invoke('show')
+        .click();
+    // Testing for errors found, rgb(164, 4, 3) is the red border color
+    cy.get('[data-cy="plugin-properties-validate-btn"]').click();
+    cy.get('[data-cy="plugin-properties-errors-found"]')
+        .should('have.text', '3 errors found.');
+    cy.get('[data-cy="referenceName"] > [data-cy="widget-wrapper-container"]')
+        .should('have.css', 'border-color', 'rgb(164, 4, 3)');
+    // Testing if the widget is highlighted when there is an error
+    // and the error text
+    cy.get('[data-cy="referenceName"] ~ [data-cy="property-row-error"]')
+        .should(
+            'have.text', 'Required property \'referenceName\' has no value.');
+    cy.get('[data-cy="referenceName"] > [data-cy="widget-wrapper-container"]')
+        .should('have.css', 'border-color', 'rgb(164, 4, 3)');
```

Comment:
nit: One suggestion would be to make this into a variable. Tomorrow if we decided to change the shade of red this would need to be updated in more places if we have more tests testing for color.

Reply:
Fixed